File: 26-rmv-dups-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        testElement = nums[0]
        unique_element_it = 1
        for index, current_element in enumerate(nums):
            if current_element != testElement:
                if index != unique_element_it:
                    nums[unique_element_it] = current_element
                testElement = current_element
                unique_element_it += 1
        return unique_element_it

File: 26-rmv-dups-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int myElement = nums[0];
        int my_non_dup_list_it = 1;
        for (int i = 1; i < nums.size(); ++i)
        {
            if (nums[i] != myElement)
            {
                if (my_non_dup_list_it != i)
                {
                    nums[my_non_dup_list_it] = nums[i];
                }
                myElement = nums[i];
                my_non_dup_list_it++;
            }
            
        }
        return my_non_dup_list_it;
    }
};